src/watchdog_fsevents.c

```diff
@@ -798,7 +837,10 @@ static PyObject *
 watchdog_stop(PyObject *self, PyObject *emitter_thread)
 {
     UNUSED(self);
-    PyObject *value = PyDict_GetItem(thread_to_run_loop, emitter_thread);
+    PyObject *value = NULL;
+    LOCK();
+    PyDict_GetItemRef(thread_to_run_loop, emitter_thread, &value);
+    UNLOCK();
```

Comment:
Same issue

Reply:
Also you have to check the return value of `PyDict_GetItemRef` and possibly handle errors.